GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -3218,6 +3220,45 @@ private void FormBrowse_DragEnter(object sender, DragEventArgs e)
             }
         }
 
+        private void JumpToArtificialCommit()
+        {
+            var selectedGuid = RevisionGrid.LatestSelectedRevision.Guid;
+            string guidToSelect;
+            switch (selectedGuid)
+            {
+                case GitRevision.WorkTreeGuid: guidToSelect = GitRevision.IndexGuid; break;
+                case GitRevision.IndexGuid: guidToSelect = GitRevision.WorkTreeGuid; break;
+                default:
+                    if (!RevisionGrid.GetChangeCount(ObjectId.WorkTreeId).IsEmpty)
+                    {
+                        guidToSelect = GitRevision.WorkTreeGuid;
+                    }
+                    else if (!RevisionGrid.GetChangeCount(ObjectId.IndexId).IsEmpty)
+                    {
+                        guidToSelect = GitRevision.IndexGuid;
+                    }
+                    else
+                    {
+                        guidToSelect = RevisionGrid.GetRevision(ObjectId.IndexId).ParentIds.FirstOrDefault()?.ToString();
+#if false
+                        if (guidToSelect == selectedGuid)
+                        {
+                            guidToSelect = GitRevision.WorkTreeGuid;
+                        }
+#endif
+                    }
+
+                    break;
+            }
+
+            GoToRef(guidToSelect, showNoRevisionMsg: false);
+            if (revisionDiff.Visible)
```

Comment:
I would prefer to focus the difftab if another tab is in focus

Reply:
if (!revisionDiff.Focused)
            {
                CommitInfoTabControl.SelectedTab = DiffTabPage;
            }

Reply:
Focusing the FileStatusList does not work reliably because the list is empty yet due to slow loading.

Reply:
There seems to be a delay before artificial commits are loaded or else it does take longer than for real commits.